src/librustc_mir/transform/const_prop.rs

```diff
@@ -857,22 +889,32 @@ impl<'mir, 'tcx> MutVisitor<'tcx> for ConstPropagator<'mir, 'tcx> {
                 if let Some(local) = place.as_local() {
                     let can_const_prop = self.can_const_prop[local];
                     if let Some(()) = self.const_prop(rval, place_layout, source_info, place) {
-                        if can_const_prop == ConstPropMode::FullConstProp
-                            || can_const_prop == ConstPropMode::OnlyPropagateInto
-                        {
+                        if can_const_prop != ConstPropMode::NoPropagation {
+                            // This will return None for Locals that are from other blocks,
+                            // so it should be okay to propagate from here on down.
                             if let Some(value) = self.get_const(local) {
                                 if self.should_const_prop(value) {
                                     trace!("replacing {:?} with {:?}", rval, value);
                                     self.replace_with_const(rval, value, statement.source_info);
-
                                     if can_const_prop == ConstPropMode::FullConstProp {
+                                        // FIXME(felix91gr): is this trace accurate??
+                                        // See: next FIXME
```

Comment:
"next FIXME" is a bit vague, there's nothing in the next 50 lines of code, can you write this comment in a way that will survive refactorings?

Reply:
Oh yes, sorry, it's indeed outdated and I've actually just removed it because the issue has been solved :D